WordPress/gutenberg
# Framework: Specify missing wp-url dependencies

Related: #8300

This pull request seeks to resolve an issue where the Archives block produces an error. This was caused by #8300, where new dependencies were introduced on the `@wordpress/url` module, but not specified in `client-assets.php`. The changes here add this dependency to `wp-core-blocks` and `wp-components` script handles. While I was here, I also re-alphabetized the dependency list for these handles (and for `wp-editor`, which was touched in #8300 but already defined `wp-url` as a dependency).

In the future, we might consider how to automate or validate this process, so that it's not manually curated. I had taken an attempt here, first with a plain PHP script, then with a custom PHPCS Sniff. Ultimately, `gutenberg_register_scripts_and_styles` touches many core functions which need to be defined, meaning that it's most likely needed to load most of the WordPress context to do programatically. Another alternative is to run validate at runtime (at least in `SCRIPT_DEBUG`) to check that, for any package defined in `packages/`, its corresponding script registration includes all the same dependencies as listed in its `package.json` `dependencies` object.

**Testing instructions:**

1. Navigate to Posts > Add New
2. Insert an Archives block
3. Verify there are no errors
